mail_tracking/static/src/js/failed_message/widget.js

```diff
@@ -0,0 +1,311 @@
+            } else {
+                this.$el.empty();
+            }
+        },
+
+        /**
+        * Reset widget data using selected record
+        *
+        * @private
+        * @param {Object} record
+        */
+        _reset: function (record) {
+            this._super.apply(this, arguments);
+            this.failed_messages = this.record.specialData[this.name];
+            this.res_id = record.res_id;
+        },
+
+        /**
+        * Trigger event to reload mail widgets
+        *
+        * @private
+        * @param {Array} fieldsToReload
+        */
+        _reload: function (fieldsToReload) {
+            this.trigger_up('reload_mail_fields', fieldsToReload);
+        },
+
+        /**
+        * Mark failed message as reviewed
+        *
+        * @private
+        * @param {Int} id
+        * @returns {Promise}
+        */
+        _markFailedMessageReviewed: function (id) {
+            return this._rpc({
+                model: 'mail.message',
+                method: 'toggle_tracking_status',
+                args: [[id]],
+                context: this.record.getContext(),
+            });
+        },
+
+        // Handlers
+        /**
+        * Listen bus notification to launch reload process.
+        * This bus notification is received when the user uses
+        * 'mail.resend.message' wizard.
+        *
+        * @private
+        * @param {Array} notifs
+        */
+        _onNotification: function (notifs) {
+            var self = this;
+            _.each(notifs, function (notif) {
+                var model = notif[0][1];
+                if (model === 'res.partner') {
+                    var data = notif[1];
+                    if (data.type === 'update_failed_messages') {
+                        // Reload 'mail_failed_message' widget
+                        self._reload({failed_message: true});
+                    }
+                }
+            });
+        },
+
+        /**
+         * Handle retry failed message event to open the mail.resend.message
+         * wizard.
+         *
+         * @private
+         * @param {Event} event
+         */
+        _onRetryFailedMessage: function (event) {
+            event.preventDefault();
+            var messageID = $(event.currentTarget).data('message-id');
+            this.do_action('mail.mail_resend_message_action', {
+                additional_context: {
+                    mail_message_to_resend: messageID,
+                },
+            });
+        },
+
+        /**
+         * Handle mark message as reviewed event
+         *
+         * @private
+         * @param {Event} event
+         */
+        _onMarkFailedMessageReviewed: function (event) {
+            event.preventDefault();
+            var messageID = $(event.currentTarget).data('message-id');
+            this._markFailedMessageReviewed(messageID).then(
+                this._reload.bind(this, {failed_message: true}));
```

Comment:
Lazy is better here too:

```suggestion
                $.proxy(this, "_reload", {failed_message: true}));
```